src/core/DataFile.cpp

```diff
@@ -163,6 +163,53 @@ DataFile::~DataFile()
 
 
 
+bool DataFile::validate( QString extension )
+{
+	switch( m_type )
+	{
+	case Type::SongProject:
+		if( extension == "mmp" || extension == "mmpz" )
+		{
+			return true;
+		}
+		break;
+	case Type::SongProjectTemplate:
+		if(  extension == "mpt" )
+		{
+			return true;
+		}
+		break;
+	case Type::InstrumentTrackSettings:
+		if ( extension == "xpf" || extension == "xml" )
+		{
+			return true;
+		}
+		break;
+	case Type::UnknownType:
+		if (! ( extension == "mmp" || extension == "mpt" || extension == "mmpz" ||
```

Comment:
> Are you suggesting something like the following

Yeah, similar.  It would be more like this:

```cpp
switch (type) {
    case Project:
    case ProjectCompressed:
    case ProjectTemplate:
    case ZynPreset:
    // etc
}
```